hibernate-reactive-core/src/main/java/org/hibernate/reactive/adaptor/impl/ResultSetAdaptor.java

```diff
@@ -430,6 +430,8 @@ public String getCatalogName(int column) {
 
 			@Override
 			public String getColumnTypeName(int column) {
+				// This information is in rows.columnDescriptors().get( column-1 ).dataType.name
+				// but does not appear to be accessible.
```

Comment:
@gbadner this is used by `SchemaValidator`, is that right?

Should we ask the vert.x guys to expose this info on `ColumnDescriptor`?

Reply:
@gavinking,

TL; DR: yes, it would help to expose this info on `ColumnDescriptor`.

For the record, here are the details:

If they expose it, then we can use `ResultSetMetaData` like what is done in ORM [here](https://github.com/hibernate/hibernate-orm/blob/a64d0102812474898059f1143bf09764f95556f1/hibernate-core/src/main/java/org/hibernate/tool/schema/extract/internal/InformationExtractorJdbcDatabaseMetaDataImpl.java#L152-L167) in the ORM [PR](https://github.com/hibernate/hibernate-orm/pull/4106).

Currently, since that information is not available in `ResultSetMetaData`, [`AbstractReactiveInformationSchemaBasedExtractorImpl#processColumnsResultSet`](https://github.com/hibernate/hibernate-reactive/pull/903/files#diff-149c9cc5fe97eafff78515120d487c5934371492bd13929253d0ec88278e9316R220-R259) is executed instead. 

Notice also that the SQL type code gets returned as `null` by `AbstractReactiveInformationSchemaBasedExtractorImpl#processColumnsResultSet` (which ends up being 0 (zero), which is java.sql.Types.NULL) because it is not available. It would be helpful to get that SQL code; otherwise, this [line](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/tool/schema/internal/AbstractSchemaValidator.java#L165) constructing an error message might not make sense (or may throw an exception?).

From looking at the code, it appears that `ResultSetAdapter`'s implementation of `ResultSetMetaData#getColumnType` returns the SQL type code properly, but I haven't debugged into it to be sure.

AFAICT, `ColumnInformation#getNullable`, `#getColumnSize`, and `#getDecimalDigits` are not used by ORM's main branch, so if `ColumnDescriptor#dataType#name` gets exposed, we'll have everything ORM currently needs. I know that there has been talk about using some of this information in the future (in ORM 6?), but I'm not sure about that.

Reply:
I opened https://github.com/eclipse-vertx/vertx-sql-client/issues/1020.